lib/msf/core/rpc/v10/service.rb

```diff
@@ -140,11 +140,16 @@ def process(req)
         end
       end
 
-      ::Timeout.timeout(self.dispatcher_timeout) { self.handlers[group].send(mname, *msg) }
+      ::Timeout.timeout(self.dispatcher_timeout) do
+        Thread.current[:rpc_token] = token
+        self.handlers[group].send(mname, *msg)
+      end
 
     rescue ::Exception => e
       elog('RPC Exception', error: e)
       process_exception(e)
+    ensure
+      Thread.current[:rpc_token] = nil
```

Comment:
if `elog` or `process_exception` raises - then I think `Thread.current[:rpc_token]` would end up accidentally being stale here

Reply:
This is fake news; will resolve 💯